### PR TITLE
Add try-runtime to ci pipeline

### LINKED ISSUE #1617: GHA pipeline features
### Description
Issue to define the pipeline enchancements we have discussed within the team that are "nice to have":
- [x] ~~Add `try_runtime` https://paritytech.github.io/try-runtime-cli/try_runtime/:~~
    ~~- When label `D8-migration` exists in PR~~
    ~~- Weekly on the main branch for sanity~~
   Solved by: #1735
- [x] Add code coverage to PR sanity checks (ideally per pallet).
   Solved by: #1745
   [ ] Pending: add integration tests to the tarpaulin code coverage, currently it hangs forever.
- [ ] Code scanning and security scanning (clippy, [miri](https://github.com/rust-lang/miri), [MIRAI](https://github.com/facebookexperimental/MIRAI)
- [ ] Test running srtool as a workflow container and setup sccache (the GHA cache is 2GB, too big)
- [ ] Post/upload srtool summary
- [ ] Review the Dockerfile again 
- [ ] Publish binary when release created
- [ ] Update the DockerHub Overview with a dynamic Readme file https://github.com/peter-evans/dockerhub-description
- [ ] Verifying all crates have proper Cargo features set via `subalfred`
- [ ]  (Optional) Add timing reports https://doc.rust-lang.org/nightly/cargo/reference/timings.html
- [ ] Fuzz testing?
- [ ] Ping Devs in Slack

### Research/based on

https://centrifuge.hackmd.io/Ftgot4ilSxK5ERyrEujnBg

### PR DESCRIPTION
**Pull request by gpmayorga**
# Description

Add try-runtime for Altair and Centrifuge to CI

Fixes  #1617

## Changes and Descriptions

Add try-runtime to PRs labeled as D8-migration

### CODE REVIEW

File: .github/workflows/try-runtime.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    name: try-runtime "${{ matrix.chain }}"
+    steps:
+      - name: Check out code
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab #3.5.2
+
+      - name: Prep build on Ubuntu
+        uses: ./.github/actions/prep-ubuntu 
+
+      - name: Install try-runtime-cli
+        run: cargo install --git https://github.com/paritytech/try-runtime-cli --tag v0.5.4 --locked
```
Comment (lemunozm) [inline .github/workflows/try-runtime.yml:25]: Why is this needed?

**Review by lemunozm — APPROVED** [review]
Straight to the point!

Maybe could be interesting to add the `try-runtime` command as an script in `run-check.sh` to launch in local easily for us. WDYT? No blocker anyways

File: .github/workflows/try-runtime.yml (hunk 1 above)
Comment (gpmayorga) [reply .github/workflows/try-runtime.yml:70]: prep-ubuntu peforms `scripts/install_toolchain.sh`. Is that enough ?
I'll try removing it

**Comment by gpmayorga** [comment]
> /try-runtime

Unfortunately until this is merged to main I can't test this trigger

**Comment by gpmayorga** [comment]
Here's the job with the latest code:
https://github.com/centrifuge/centrifuge-chain/actions/runs/7932658994/job/21659621848

It seems to be running fine (it fails where it should). Feel free to add the D8-migration tag if you want to see the job running. See it running at https://github.com/centrifuge/centrifuge-chain/actions/workflows/try-runtime.yml

**Comment by lemunozm** [comment]
I would say, wait until https://github.com/centrifuge/centrifuge-chain/issues/1641 is fixed to not break all PR that contains migrations. It should be fixed before the next RU, so maybe 1-2 weeks until merging this PR. Is it ok?

**Comment by wischli** [comment]
> I would say, wait until #1641 is fixed to not break all PR that contains migrations. It should be fixed before the next RU, so maybe 1-2 weeks until merging this PR. Is it ok?

Unfortunately, we need to merge #1641 and upgrade all chains including the fix before we can expect `try-runtime` to work. I want to include these fixes into RU 1025 such that 2 weeks seems reasonable.

**Review by lemunozm — APPROVED** [review]
LGTM!

**Review by wischli — APPROVED** [review]
Thanks for another CI improvement!
